### PR TITLE
LPS-156222 | master

### PR DESCRIPTION
**Pull request by sontruongces**
Hi team,
This PR is the fix for the [LPS-156222](https://issues.liferay.com/browse/LPS-156222), the original has passed my review at sontruongces#20. Please help me review it and give me feedback.
Thanks!
cc: @locpham97

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/localizable/text/LocalizableTextDDMFormFieldValueAccessor.java [hunk 1]
```diff
@@ -82,7 +84,10 @@ public JSONObject getValueForEvaluation(
 	public boolean isEmpty(DDMFormFieldValue ddmFormFieldValue, Locale locale) {
 		JSONObject jsonObject = getValue(ddmFormFieldValue, locale);
 
-		if (jsonObject.length() == 0) {
+		if ((jsonObject.length() == 0) ||
+			Validator.isNull(
+				jsonObject.get(LanguageUtil.getLanguageId(locale)))) {
```
[the remark this answers is not in the document]
Comment (sontruongces) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/localizable/text/LocalizableTextDDMFormFieldValueAccessor.java]: @drewbrokke, the fix was updated based on your comment. Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/localizable/text/LocalizableTextDDMFormFieldValueAccessor.java [hunk 2]
```diff
@@ -86,7 +88,15 @@ public boolean isEmpty(DDMFormFieldValue ddmFormFieldValue, Locale locale) {
 			return true;
 		}
 
-		return false;
+		Set<String> jsonObjectKeys = jsonObject.keySet();
+
+		for (String key : jsonObjectKeys) {
```
Comment (drewbrokke) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/java/com/liferay/dynamic/data/mapping/form/field/type/internal/localizable/text/LocalizableTextDDMFormFieldValueAccessor.java:93]: SF: Inline the `jsonObjectKeys` variable.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@locpham97 overall this looks good. However, we also need to add test coverage for this scenario. Can you please include an assertion inside of [LocalizableTextDDMFormFieldValueAccessorTest.java](https://github.com/drewbrokke/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/localizable/text/LocalizableTextDDMFormFieldValueAccessorTest.java#L69-L81)? Thank you!

**Comment by sontruongces** [comment]
@drewbrokke , the fix was SF-ed and added test by @locpham97. Thanks!

**Comment by drewbrokke** [comment]
See https://github.com/liferay-forms/liferay-portal/pull/1896. Thanks!

**Pull request closed without merging**